datalad/metadata/search.py

```diff
@@ -173,16 +175,48 @@ def _deep_kv(basekey, dct):
             else:
                 yield key, v
 
+    def get_indexer_for(metadata_format_name: str) -> callable:
+        indexer = (tuple(iter_entry_points('datalad.metadata.indexers', metadata_format_name)) or (None,))[0]
```

Comment:
We only look at the first returned entry point, if any, because there should be at most one indexer for ``metadata_format_name´´.

Comment added.

Reply:
I would prefer such assumption "coded in" with the feedback to the "user".  E.g. at least a log debug message (or better a warning) issued if more than one somehow found present.  It is often such assumptions which later cause a big grief trying to figure out WTF my X is not working (because you had multiple but some other was chosen and nothing alerted you about that)

Reply:
The setuptools do not allow duplicated names in a group. So to get multiple entry points, you would have to do something very weird. ThereforeI I assumed that there would only be one entry_point for the given name and group.

If the name is not present in the group, an appropriate error is shown to the user.

Reply:
https://setuptools.readthedocs.io/en/latest/pkg_resources.html
doc says
```
iter_entry_points(group, name=None)
Yield entry point objects from group matching name

If name is None, yields all entry points in group from all distributions in the working set, otherwise only ones matching both group and name are yielded. ...
```

I have not tried to overload etc, but insofar documentation hints that there could be multiple entry points for the same (group, name).  Overall, I would not rely on some assumption whenever function is an iterator "by design" -- why not to "hard-code" the assumption and make it explicit here?

Reply:
This [doc](https://setuptools.readthedocs.io/en/latest/pkg_resources.html?highlight=iter_entry_points#entry-points) says:
```
Each project distribution can advertise at most one entry point of a given name within the same entry point group. For example, a distutils extension could advertise two different distutils.commands entry points, as long as they had different names.
```
And indeed install reports an error from setup when duplicating entry points:
```
error in ... setup command: ('Duplicate entry point', 'datalad.metadata.indexers', 'metalad_studyminimeta')
```
I don't want to circle too long on this one. We could add an assert to ensure that there is only one entry-point of the given group and name (although that should be guaranteed by setuptools)?

Reply:
Issue created: https://github.com/datalad/datalad/issues/5043

Reply:
effectively current code "allows" for multiple implementations, but silently disregards all but first. #5043 is indeed good to solve but I would just insist on adding at least a `lgr.debug` or better `lgr.warning` (since unsupported until decision on #5043 is made) if multiple are detected.

Reply:
Added a warning and fall back to built-in indexer, if multiple indexers are found.